lib/isomorphic/create-routes.coffee

```diff
@@ -12,6 +13,15 @@ module.exports = (pages, pagesReq) ->
     handler: pagesReq './_template'
   })
 
+  # Adds a handler for <NotFoundRoute /> that will serve as a 404 Page
+  if config.notFound
+    templates.rootNotFound = Router.createNotFoundRoute({
```

Comment:
The removal of NotFound https://github.com/rackt/react-router/blob/master/UPGRADE_GUIDE.md#notfound-route could be relevant here.